Unable to load a wallpaper

### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/iamrasel/lunar-launcher/issues) or [closed](https://github.com/iamrasel/lunar-launcher/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.

### App version

2.2.2

### Android version

13

### Steps to reproduce the bug

click the change background button and screen just flashes

### Expected behavior

to pop up to change the wallpaper

### Screenshots/Screen recordings

_No response_

### Logs

_No response_

### Additional information

setting a wallpaper with the default way with google pixel 7 then changing the background of the app to be transparent does work though.

```
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err: java.io.IOException: Cannot run program "su": error=13, Permission denied
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.ProcessBuilder.start(ProcessBuilder.java:1050)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.Runtime.exec(Runtime.java:694)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.Runtime.exec(Runtime.java:524)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.Runtime.exec(Runtime.java:421)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at rasel.lunar.launcher.helpers.UniUtils$Companion.isRooted(UniUtils.kt:114)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at rasel.lunar.launcher.settings.childs.Misc.onCreateView(Misc.kt:77)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2963)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.DialogFragment.performCreateView(DialogFragment.java:489)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:518)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
2023-01-03 13:59:13.944 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at android.os.Handler.handleCallback(Handler.java:942)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:99)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at android.os.Looper.loopOnce(Looper.java:201)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at android.os.Looper.loop(Looper.java:288)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:7872)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err: Caused by: java.io.IOException: error=13, Permission denied
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.UNIXProcess.forkAndExec(Native Method)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.UNIXProcess.<init>(UNIXProcess.java:133)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.ProcessImpl.start(ProcessImpl.java:141)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err:     at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
2023-01-03 13:59:13.945 8120-8120/rasel.lunar.launcher.debug W/System.err: 	... 21 more
```

--- a/app/src/main/kotlin/rasel/lunar/launcher/settings/childs/Appearances.kt
+++ b/app/src/main/kotlin/rasel/lunar/launcher/settings/childs/Appearances.kt
@@ -19,6 +19,7 @@
 package rasel.lunar.launcher.settings.childs
 
 import android.Manifest.permission.READ_EXTERNAL_STORAGE
+import android.Manifest.permission.READ_MEDIA_IMAGES
 import android.app.Activity.RESULT_OK
 import android.app.AlertDialog
 import android.app.WallpaperManager

--- a/app/src/main/kotlin/rasel/lunar/launcher/settings/childs/Appearances.kt
+++ b/app/src/main/kotlin/rasel/lunar/launcher/settings/childs/Appearances.kt
@@ -121,10 +121,19 @@ internal class Appearances : BottomSheetDialogFragment() {
     }
 
     private fun selectWallpaper() {
-        if (requireActivity().checkSelfPermission(READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
-            requireActivity().requestPermissions(arrayOf(READ_EXTERNAL_STORAGE), 1)
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU) {
+            // only for TIRAMISU and newer versions
+            if (requireActivity().checkSelfPermission(READ_MEDIA_IMAGES) != PackageManager.PERMISSION_GRANTED) {
+                requireActivity().requestPermissions(arrayOf(READ_MEDIA_IMAGES), 1)
+            } else {
+                wallpaperChangeLauncher.launch(Intent(Intent.ACTION_PICK).setType("image/*"))
+            }
         } else {
-            wallpaperChangeLauncher.launch(Intent(Intent.ACTION_PICK).setType("image/*"))
+            if (requireActivity().checkSelfPermission(READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                requireActivity().requestPermissions(arrayOf(READ_EXTERNAL_STORAGE), 1)
+            } else {
+                wallpaperChangeLauncher.launch(Intent(Intent.ACTION_PICK).setType("image/*"))
+            }
         }
     }
 

